src/platform/packages/shared/kbn-esql-validation-autocomplete/src/autocomplete/commands/join/index.ts

```diff
@@ -66,34 +90,14 @@ const suggestFields = async (
     getVariablesByType
   );
 
-  const intersection = suggestionIntersection(joinFields, sourceFields);
+  const commonFields = suggestionIntersection(joinFields, sourceFields);
   const union = suggestionUnion(sourceFields, joinFields);
 
-  for (const commonField of intersection) {
-    commonField.sortText = '1';
-    commonField.documentation = {
-      value: i18n.translate('kbn-esql-validation-autocomplete.esql.autocomplete.join.sharedField', {
-        defaultMessage: 'Field shared between the source and the lookup index',
-      }),
-    };
-
-    let detail = commonField.detail || '';
-
-    if (detail) {
-      detail += ' ';
-    }
-
-    detail += i18n.translate(
-      'kbn-esql-validation-autocomplete.esql.autocomplete.join.commonFieldNote',
-      {
-        defaultMessage: '(common field)',
-      }
-    );
-
-    commonField.detail = detail;
+  for (const commonField of commonFields) {
+    attachCommonFieldMetadata(commonField);
   }
 
-  return [...intersection, ...union];
+  return suggestionUnion(commonFields, union);
```

Comment:
Key change is here. This fixes a bug where common fields were shown twice in the JOIN suggestions list.